backend/lcfs/web/api/compliance_report/update_service.py

```diff
@@ -213,9 +220,11 @@ async def handle_assessed_status(self, report: ComplianceReport):
         )
         if not has_director_role:
             raise HTTPException(status_code=403, detail="Forbidden.")
-        # Update the transaction to assessed
-        report.transaction.transaction_action = TransactionActionEnum.Adjustment
-        report.transaction.update_user = user.keycloak_username
+
+        if report.transaction:
+            # Update the transaction to assessed
+            report.transaction.transaction_action = TransactionActionEnum.Adjustment
+            report.transaction.update_user = user.keycloak_username
```

Comment:
it would be good idea to raise Value error here.
else:
    # Handle the case where the transaction is missing
    raise ValueError(
        f"Transaction is missing for the compliance report with ID: {report.compliance_report_id}."
    )

Reply:
For 0 credit compliance reports we no longer create a transaction, I talked to Alex and we agreed that made more sense than making a 0 credit adjustment, so not all reports will have transactions.